frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.styled.ts

```diff
@@ -43,11 +44,16 @@ export const ParameterContainer = styled.div<ParameterContainerProps>`
   border-radius: 0.5rem;
   cursor: pointer;
   font-weight: bold;
-  width: 170px;
+  min-width: 170px;
   margin-right: 0.5rem;
   margin-bottom: 0.5rem;
   padding: 0.5rem;
   color: ${props => props.isEditingParameter && color("white")};
   background-color: ${props =>
     props.isEditingParameter ? color("brand") : color("white")};
 `;
+
+export const SettingsIcon = styled(Icon)`
+  margin-left: auto;
+  padding-left: 1rem;
+`;
```

Comment:
Giving the icon some minimal left padding to avoid this:

<img width="480" alt="no-icon-padding" src="https://github.com/metabase/metabase/assets/17258145/89151a91-15a7-4a2c-96a3-f4c8b3c86014">